### PR TITLE
for single-run, print progress messages and start/stop stats

### PR DESCRIPTION
**Pull request by bhoflich**
Signed-off-by: Benjamin Hoflich <45051453+bhoflich@users.noreply.github.com>

### CODE REVIEW

File: runHarness/WorkloadDrivers/AuctionWorkloadDriver.pm [hunk 1]
```diff
@@ -1174,19 +1186,87 @@ sub startRun {
...
+
 	while (!$runCompleted) {
 		$logger->debug("Sending get to $url");
 		$req = HTTP::Request->new( GET => $url );
 		$res = $ua->request($req);
 		$logger->debug(
 			"Response status line: " . $res->status_line . " for url " . $url );
 		if ( $res->is_success ) {
-			$endRunStatus = $json->decode( $res->content );	
+			$logger->debug("Response content: " . $res->content );
```
Comment (hrosenbe) [inline runHarness/WorkloadDrivers/AuctionWorkloadDriver.pm]: Can you should take this out.  I had put this when I was looking at the returned content, but it really fills up the debug log.

File: runHarness/WorkloadDrivers/AuctionWorkloadDriver.pm [hunk 2]
```diff
@@ -1174,19 +1186,87 @@ sub startRun {
...
+					$logger->debug("Workload $workloadNum, AppInstance: $wkldName: curInterval = $curIntervalName");
+				}
+				if ($usingFindMaxLoadPathType) {
+					$wkldName =~ /appInstance(\d+)/;
+					my $appInstanceNum = $1;
+
+					if (defined $statsRunning[$appInstanceNum] && $statsRunning[$appInstanceNum]) {
+						$statsRunning[$appInstanceNum] = 0;
+						$console_logger->info("   [$wkldName] Stopping performance statistics on workload.");
+						$self->workload->stopStatsCollection($tmpDir);
```
Comment (hrosenbe) [inline runHarness/WorkloadDrivers/AuctionWorkloadDriver.pm]: Just trying to understand what is happening here.  Are you calling startStats at the start, and stopStats at the end, of each VERIFYMAX iteration?  If so, I like that approach.

File: runHarness/WorkloadDrivers/AuctionWorkloadDriver.pm [hunk 3]
```diff
@@ -1174,19 +1186,87 @@ sub startRun {
 	my $endRunStatus = "";
 	my $endRunStatusRaw = "";
 	$url = $self->getControllerURL() . "/run/$runName/state";
+	my @curIntervalNames;
+	my @lastIntervalNames;
+	my @statsRunning;
+
 	while (!$runCompleted) {
```
Comment (hrosenbe) [inline runHarness/WorkloadDrivers/AuctionWorkloadDriver.pm]: One more thing that I just saw in testing with multiple appInstances.  Running with 4 appInstances, one of the appInstances hadn't started on the first fetch of the state, and so I saw:
```
Mon Jun 10 13:33:30 2019:    [appInstance1] Start: initial ramp at 200 users, duration:60s.
Mon Jun 10 13:33:30 2019:    [appInstance2] Start: initial ramp at 200 users, duration:60s.
Mon Jun 10 13:33:30 2019:    [appInstance3] Start: initial ramp at 200 users, duration:60s.
Mon Jun 10 13:33:30 2019:    [appInstance4] Start: initial ramp at 0 users, duration:60s.
```

Can you please add a sleep (say 30 seconds) before entering this loop.

File: runHarness/WorkloadDrivers/AuctionWorkloadDriver.pm (hunk 2 above)
Comment (bhoflich) [reply runHarness/WorkloadDrivers/AuctionWorkloadDriver.pm]: Basically.  Calling startStats when the poll sees a new VERIFYMAX.  Calling stopStats one poll later (so 60 seconds).  Happens for each VERIFYMAX interation, so each of start and stop will be called at least twice.

Comment (hrosenbe) [reply runHarness/WorkloadDrivers/AuctionWorkloadDriver.pm]: Sounds good to me.

**Pull request merged**